### PR TITLE
Puppi MET and MET significance fixes in the MET tool (80X)

### PR DESCRIPTION
**Pull request by mmarionncern**
Same as #15565 for 80X

I will be out of office from August 27th to September 17th. If comments comes during that period, please contact @mariadalfonso and @zdemirag .

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
the errors are 
```
No data of type "JME::JetResolutionObject" with label "AK4PFPuppi_pt" in record "JetResolutionRcd"
```
it looks like the 80X GTs do not have the needed payloads at all; in 81X version of the PR only run2 50ns GT doesn't have them.
@mmusich @franzoni  I suppose a request for updates are on the way, please post here when GT has the changes.

**Comment by mmusich** [comment]
A bit unfortunate not having the update planned ahead the PR

**Comment by slava77** [comment]
@mmusich @mariadalfonso 
what is the status of GT in 80X regarding needs for this PR ?
At least operationally, the 81X version has now settled.

**Comment by slava77** [comment]
backport of #15565

**Comment by mmusich** [comment]
@slava77 there is no update relevant yet in 80x available. 
It's in the pipeline with lower priority with respect to fixing https://github.com/cms-sw/cmssw/pull/15434.
Clarify what will be the usage of this PR, for instance if there is plan to use this in the upcoming MC production and/or data reprocessing

**Comment by slava77** [comment]
On 9/1/16 5:27 AM, Marco Musich wrote:
> @slava77 <https://github.com/slava77> there is no update relevant yet in
> 80x available.
> It's in the pipeline with lower priority with respect to fixing #15434
> <https://github.com/cms-sw/cmssw/pull/15434>.
> Clarify what will be the usage of this PR, for instance if there is plan
> to use this in the upcoming MC production and/or data reprocessing

This PR ca go into 80X only with the upcoming data rereco /MC remake.

Maria should clarify how essential it is for rereco.

If it's essential, then GT update is holding this PR and, transitively,
rereco as well.

**Comment by mmusich** [comment]
@slava77 I am not saying the needed update won't be perfomed at all, just to understand how fast you need it (if the code changes are OK in the 81x version, I would assume the GT update is needed mostly to pass the tests).

**Comment by slava77** [comment]
@mmusich 
OK if it's coming soon (this week?).
It would help to have fewer balls in the air (PRs waiting for some "external" factors to settle).

**Comment by mmusich** [comment]
Hi @slava77 the issue is that the GT queues are already ahead of the current 80X HEAD (because of #15434 - not merged). It will require a bit of time and work to reproduce the state of the current HEAD and apply on top of that the missing JME conditions. Hopefully can do it by tomorrow.

**Comment by mariadalfonso** [comment]
@mmusich @slava77 @cvuosalo 
Hi Slava we need this PR.
I will check for update from @mmusich tomorrow morning
This likely need to be rebased after the #15616 is merged (locally I need to resolve conflicts), I will check is in synch with the 81X #15565

**Comment by mmusich** [comment]
@mariadalfonso  https://github.com/mmusich/cmssw/commit/8d72e604bdc884d4dcc4468127f91a564e6fbd0a has hopefully the content you need

**Comment by mmusich** [comment]
please test

**Comment by mariadalfonso** [comment]
@mmusich added GT commit
@slava77 this is now in synch with the current #15565 (81X version)

File: PhysicsTools/PatAlgos/test/corMETFromMiniAOD.py [hunk 1]
```diff
@@ -75,9 +75,9 @@
 ### =====================================================================================================
 # Define the input source
 if runOnData:
-  fname = 'root://eoscms.cern.ch//store/relval/CMSSW_8_0_0_pre4/SinglePhoton/MINIAOD/80X_dataRun2_v0_RelVal_sigPh2015D-v1/00000/600919D1-51AA-E511-8E4C-0025905B855C.root'
+  fname = 'root://eoscms.cern.ch//store/relval/CMSSW_8_1_0_pre10/DoubleEG/MINIAOD/81X_dataRun2_relval_v4_RelVal_doubEG2015D-v1/00000/88859857-6168-E611-9E79-0025905A60AA.root'
```
Comment (slava77) [inline PhysicsTools/PatAlgos/test/corMETFromMiniAOD.py]: were you actually able to run this in 80X?
I think that 80X test should be reading 80X, not 81X.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/15600 eae5dbb
- backport of MET features from 81X: this is a combination of #15565 and older features merged in 81X much earlier. Most non-trivial edits are in PhysicsTools/PatUtils/python/tools/runMETCorrectionsAndUncertainties.py: the version of this file from this PR becomes the same as 81X (after #15565)
- jenkins tests pass and comparisons with baseline show differences consistent with 81X: in puppiMET (monitored in DQM). 
- local tests confirm behavior of jenkins tests and also show now filled MET significance. The technical performance changes are similar to 81X (roughly the same total CPU and small increase in miniAOD size).

**Comment by mmusich** [comment]
+1

**Comment by slava77** [comment]
hold

a formal hold to indicate that changes should go in to the rereco release, not sooner.

**Comment by slava77** [comment]
@davidlange6 
just as head up: this PR broke data processing configs.
I'll investigate and see if I can come up with a quick fix.

**Comment by slava77** [comment]
#15791 fixes the issue introduced by this PR in DataProcessing configs

**Pull request merged**